code/modules/atmospherics/gasmixtures/gas_mixture.dm

```diff
@@ -410,34 +410,36 @@ GLOBAL_LIST_INIT(gaslist_cache, init_gaslist_cache())
 	if(!length(cached_gases))
 		return
 	var/list/reactions = list()
-	for(var/I in cached_gases)
-		reactions += SSair.gas_reactions[I]
-	if(!length(reactions))
-		return

```

Comment:
I actually think the most elegant solution is to get rid of your changes in this file, and insert this right here:
```DM
if(length(reactions) > 1)
    sortTim(reactions, /proc/cmp_gas_reaction)
```
using the original (as in, the one you restored) comparison proc, without all the extra nonsense. this will add some extra overhead but i suspect is still faster than the original code without the gas checks. my suspicion is that in the vast, vast majority of cases only 0 or 1 reactions will be possible at any time.

Reply:
note that it's probably still helpful to sort the main reactions list on SSair with the method currently used, with it being lists of lists, so that the reactions list is partially sorted before starting the final sorting pass in react itself

Reply:
obviously you'll need to profile both my suggestion as well as your current implementation in comparison with the live solution before this can be merged

Reply:
We got a few of TEMP + single gas reactions, so i'm not sure on that first conjecture. And our generic sort implementation has awful overhead on top of that.

Reply:
yeah you're right tbh it's hard to say. profiles will tell, anyway.

Reply:
Seems like the way I did it is the slowest of them.